etl/command.py

```diff
@@ -331,68 +343,80 @@ def sanity_check_db_settings() -> None:
-                    continue
+    # For export:// steps, add the grapher:// steps that are needed to upsert data to DB.
+    for step in list(dag.keys()):
+        if step.startswith("export://multidim/") or step.startswith("export://explorers/"):
+            for dep in list(dag[step]):
+                if re.match(r"^data://grapher/", dep) or re.match(r"^data-private://grapher/", dep):
+                    dag[step].add(re.sub(r"^(data|data-private)://", "grapher://", dep))
 
-                # We want to execute export steps after any grapher://grapher steps,
-                # to ensure that any indicators required by the collection step are already pushed to DB.
-                # To achieve that, replace "data://grapher" dependencies with "grapher://grapher".
-                dag[step] = {
-                    re.sub(r"^(data|data-private)://", "grapher://", dep)
-                    if re.match(r"^data://grapher/", dep) or re.match(r"^data-private://grapher/", dep)
-                    else dep
-                    for dep in dag[step]
-                }
-
-        # Finally, ensure that the added grapher://grapher steps will be executed,
-        # by activating the "grapher" flag.
-        grapher = True
-    else:
-        # If there were any "export://" steps in the dag, remove them.
-        dag = {step: deps for step, deps in dag.items() if not step.startswith("export://")}
-
-    # If --grapher is set, add all steps for upserting to DB
-    if grapher:
-        steps = _grapher_steps(dag, private)
-        dag.update(steps)
-    # Otherwise just add those in dependencies, even if they're private
-    else:
-        steps = _grapher_steps(dag, True)
-        for deps in list(dag.values()):
-            for dep in deps:
-                if dep.startswith("grapher://"):
-                    dag[dep] = steps[dep]
+    # Add all grapher steps
+    dag.update(_grapher_steps(dag, private=True))
 
     # Validate the DAG
     _check_dag_completeness(dag)
 
     return dag
 
 
-def run_dag(
+def construct_subdag(
     dag: DAG,
     includes: Optional[List[str]] = None,
+    excludes: Optional[List[str]] = None,
+    grapher: bool = False,
+    export: bool = False,
+    private: bool = False,
+    only: bool = False,
+    exact_match: bool = False,
+) -> DAG:
+    orig_includes = includes
+
+    # Include everything when no arguments provided
+    if not includes:
+        includes = [".*"]
+    if not excludes:
+        excludes = []
+
+    # Export steps
+    if not export:
+        excludes.append("export://.*")
+
+    # Grapher steps
+    if not grapher and not export:
+        excludes.append("grapher://.*")
+
+    # Exclude private steps
+    if not private:
+        excludes.append("private://.*")
+
+    # Exclude grapher regions, they're fetched by owid-grapher as CSV from catalog
+    # but are not supposed to be in DB
+    excludes.append("grapher://grapher/regions/latest/regions")
```

Comment:
This step is now archived, so I suppose we don't need this hack anymore, do we?

Reply:
:tada: